cci_cameroon/analysis/clustering_rumours/rumour_clustering_model_development.py

```diff
@@ -0,0 +1,1048 @@
+# -*- coding: utf-8 -*-
+# ---
+# jupyter:
+#   jupytext:
+#     cell_metadata_filter: -all
+#     comment_magics: true
+#     text_representation:
+#       extension: .py
+#       format_name: percent
+#       format_version: '1.3'
+#       jupytext_version: 1.13.0
+#   kernelspec:
+#     display_name: Python 3 (ipykernel)
+#     language: python
+#     name: python3
+# ---
+
+# %%
+import pandas as pd
+import matplotlib.pyplot as plt
+import numpy as np
+import cci_cameroon
+from langdetect import detect, detect_langs
+from googletrans import Translator
+from cdlib import evaluation
+
+# %matplotlib inline
+import faiss
+from faiss import normalize_L2
+from sentence_transformers import SentenceTransformer
+import networkx as nx
+import cdlib
+from cdlib import algorithms
+from cci_cameroon.getters.clustering_helper_functions import (
+    generate_colors,
+    draw_communities_graph,
+    generate_adjacency_matrix,
+    generate_adjacency_matrix2,
+    generate_edges,
+    get_metrics,
+    compute_community_silhuoette_scores,
+    get_communities_with_threshold,
+    generate_community_labels,
+)
+from sklearn.metrics import silhouette_samples, silhouette_score
+from sklearn.metrics.cluster import adjusted_mutual_info_score
+import community
+import matplotlib
+
+# %% [markdown]
+# # Approach used
+#
+# * Load and clean the dataset
+# * Convert the unlabelled comments into vector of numerical values using a transformer model
+# * Using FAISS library, compute pairwise similarity scores for the comments
+# * Based on the similarity scores, a connectivity graph
+# * Using a community detection algorithm, we generate subgroups/clusters from the graph
+# * We use modularity score to evaluate the strength of the communities formed. The larger the modularity score, the better the communities formed.
+#
+
+# %%
+project_directory = cci_cameroon.PROJECT_DIR
+
+# %%
+data_df = pd.read_excel(
```

Comment:
This and the below steps can be replaced by the pre-processed data created from the refactored code in pipeline (can be updated once merged into model pipeline)